swagger/src/test/scala/me/apidoc/swagger/SwaggerServiceValidatorSpec.scala

```diff
@@ -47,6 +39,172 @@ class SwaggerServiceValidatorSpec extends FunSpec with Matchers {
+                      path = "/pets/",
+                      description = Some("finds pets by status - as a query param"),
+                      deprecation = None,
+                      body = None,
+                      parameters = Seq(Parameter(
+                        name = "status",
+                        `type` = "enum_status",
+                        location = Query,
+                        description = None,
+                        deprecation = None,
+                        required = true,
+                        default = None,
+                        minimum = None,
+                        maximum = None,
+                        example = None)
+                      ),
+                      responses = Seq(
+                        Response(
+                          code = ResponseCodeInt(200),
+                          `type` = "[Pet]",
+                          description = Some("find pet response"),
+                          deprecation = None),
+                        Response(
+                          code = ResponseCodeOption.Default,
+                          `type` = "Error",
+                          description = Some("unexpected error"),
+                          deprecation = None)
+                      ),
+                      attributes = Nil),
+                    Operation(
+                      method = Get,
+                      path = "/pets/:status",
+                      description = Some("finds pets by status - as a path param"),
+                      deprecation = None,
+                      body = None,
+                      parameters = Seq(Parameter(
+                        name = "status",
+                        `type` = "enum_status",
+                        location = Path,
+                        description = None,
+                        deprecation = None,
+                        required = true,
+                        default = None,
+                        minimum = None,
+                        maximum = None,
+                        example = None)
+                      ),
+                      responses = Seq(
+                        Response(
+                          code = ResponseCodeInt(200),
+                          `type` = "[Pet]",
+                          description = Some("find pet response"),
+                          deprecation = None),
+                        Response(
+                          code = ResponseCodeOption.Default,
+                          `type` = "Error",
+                          description = Some("unexpected error"),
+                          deprecation = None)
+                      ),
+                      attributes = Nil)
+                  ),
+                  attributes = Seq())
+                )
+
+              service.enums.size should be(1)
+              checkEnum(service.enums(0),
+                Enum(
+                  name = "enum_status",
```

Comment:
Thanks @mbryzek for the feedback! 
So in an example like the test one where the same logical enum is repeated 3 times ([path param](https://github.com/fabiocognigni/apidoc/blob/517-swagger_enums_1/swagger/src/test/resources/petstore-enums.json#L25-L33), [query param](https://github.com/fabiocognigni/apidoc/blob/517-swagger_enums_1/swagger/src/test/resources/petstore-enums.json#L62-L70) and [model property](https://github.com/fabiocognigni/apidoc/blob/517-swagger_enums_1/swagger/src/test/resources/petstore-enums.json#L114-L116)) we want to have `3 different enum types imported into ApiDoc`; we want them collapsed into 1 Apidoc enum type only if $ref has been explicitly used in the Swagger spec to reuse the same enum definition.
I'll add an example/test for reusing an enum in Swagger with $ref.